models/Message.php

```diff
@@ -276,23 +275,12 @@ protected static function makeCacheKey()
     }
 
     /**
-     * Creates a sterile key for a message.
+     * Creates an md5 representation of the message Id
      * @param  string $messageId
      * @return string
      */
-    protected static function makeMessageCode($messageId)
+    public static function makeMessageCode($messageId)
     {
-        $separator = '.';
-
-        // Convert all dashes/underscores into separator
-        $messageId = preg_replace('!['.preg_quote('_').'|'.preg_quote('-').']+!u', $separator, $messageId);
-
-        // Remove all characters that are not the separator, letters, numbers, or whitespace.
-        $messageId = preg_replace('![^'.preg_quote($separator).'\pL\pN\s]+!u', '', mb_strtolower($messageId));
-
-        // Replace all separator characters and whitespace by a single separator
-        $messageId = preg_replace('!['.preg_quote($separator).'\s]+!u', $separator, $messageId);
-
-        return Str::limit(trim($messageId, $separator), 250);
+        return md5(trim($messageId));
```

Comment:
I wonder if the generated code should be case insensitive (as previous behavior)?
i.e. `return md5(trim(str_tolower($messageId)));`

Reply:
What would the advantage of case-insensitivity be? And could there be a use case for case-sensitivity? For example, if the original author writes a string in all caps and then later decides that it looks silly that way, so they update the string. I would think that the translators should be aware of that.

Reply:
Well, the same could be said for punctuation as well. People might have "My string!" in one place and "My string" in another place. I think that if you're going to create a change that could potentially break things, you might as well go all in so that the breaks occur on a single update. It's probably either now or never if you're going to change this.

Reply:
True, but I feel there is more potential for case sensitive ids than ones with punctuation.

In any case, we'll need @daftspunk's input on this breaking change.

Reply:
@mjauvin Just a thought. What about adding a setting somewhere to set whether the messages are case-sensitive or not? Then the default could be set to insensitive and allow the user to set it as sensitive should the case arise.